html/janus.js

```diff
@@ -1445,12 +1453,17 @@ function Janus(gatewayCallbacks) {
 						return false;
 					}
 
-					getUserMedia(
-						{audio: audioExist && isAudioSendEnabled(media), video: videoExist ? videoSupport : false},
-						function(stream) { pluginHandle.consentDialog(false); streamsDone(handleId, jsep, media, callbacks, stream); },
-						function(error) { pluginHandle.consentDialog(false); callbacks.error(error); });
+					navigator.mediaDevices.getUserMedia({
+						audio: audioExist ? audioSupport : false,
+						video: videoExist ? videoSupport : false
+					})
+					.then(function(stream) { pluginHandle.consentDialog(false); streamsDone(handleId, jsep, media, callbacks, stream); })
+					.catch(function(error) { pluginHandle.consentDialog(false); callbacks.error(error); });
+				})
```

Comment:
IMHO: The current frontend stack is pretty stable, but hard to include in new projects easily, besides hard to extend with new features / tests. I don't know what are the plans for the future, but it would be great to see janus frontend side in es6 with npm / webpack or similar structures.